common/network/services/receiver.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "common/network/services/receiver.h"
+#include "common/network/logger.h"
+#include "common/network/services/tcp_receiver.hpp"
+#include "common/network/services/udp_receiver.hpp"
+
+void receiver_service_prepare_packet(iota_packet_t* const packet,
+                                     size_t const length,
+                                     char const* const host,
+                                     uint16_t const port,
+                                     protocol_type_t const protocol) {
+  packet->length = length;
+  strcpy(packet->source.host, host);
+  packet->source.port = port;
+  packet->source.protocol = protocol;
+  packet->content[length] = '\0';
+}
+
+bool receiver_service_start(receiver_state_t* const state) {
+  if (state == NULL) {
+    return false;
+  }
+  try {
+    boost::asio::io_context ctx;
```

Comment:
who owns the thread running the context's event loop?
shouldn't the two receivers run on separate threads?